shark_turbine/kernel/wave/utils.py

```diff
@@ -531,3 +536,27 @@ def specialize_index_sequence(
             operand_map[key] = 1
             return index_seq.subs(operand_map)
     return index_seq.subs(operand_map)
+
+
+def find_index_bounds(
+    constraints: list[Constraint], index: dict[IndexExpr, IndexExpr]
+) -> Optional[list[IndexExpr]]:
+    bounds = []
+    for constraint in constraints:
+        if not isinstance(constraint, (WorkgroupConstraint, TilingConstraint)):
```

Comment:
Why do we ignore the WaveConstraints here? Does that mean that we assuming that the workgroup tile size is a multiple of the wave tile size?

Reply:
Yeah, I implicitly assumed WG size is divisible by wave size, do we have any potential examples when it's false?

Reply:
Good question. I was just thinking in terms of generality (so workgroup tile size = 27 and wave tile size = 17) but maybe its not such a common use case. We can ignore for now.